tensorlayer/files.py

```diff
@@ -761,17 +862,22 @@ def load_flickr1M_dataset(tag='sky', size=10, path="data", n_threads=50, printab
     for folder in images_folder_list[0:size * 10]:
         tmp = load_file_list(path=folder, regx='\\.jpg', printable=False)
         tmp.sort(key=lambda s: int(s.split('.')[-2]))  # ddd.jpg
-        images_list.extend([folder + '/' + x for x in tmp])
+        images_list.extend([os.path.join(folder, x) for x in tmp])
 
     # 2. tag path list
     tag_list = []
-    tag_folder_list = load_folder_list(path + "/tags")
-    tag_folder_list.sort(key=lambda s: int(s.split('/')[-1]))  # folder/images/ddd
+    tag_folder_list = load_folder_list(os.path.join(path, "tags"))
+
+    from sys import platform as _platform
```

Comment:
can we use [os.path.basename](https://docs.python.org/2/library/os.path.html#os.path.basename) instead of `if else platform`